### PR TITLE
feat: [ANDROAPP-5930] move details information to the registration form

### PR DESCRIPTION
**Pull request by xavimolloy**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://dhis2.atlassian.net/browse/ANDROAPP-

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] 11.X - 13.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialRepositoryImpl.java [hunk 1]
```diff
@@ -343,7 +343,7 @@ private FieldUiModel transform(@NonNull ProgramStageDataElement stage, DataEleme
                 optionSetConfig,
                 null,
                 null,
-                null);
+                null, null);
     }
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialRepositoryImpl.java:347]: is this a formatting typo?

File: form/src/main/java/org/dhis2/form/ui/provider/inputfield/DateProvider.kt [hunk 2]
```diff
@@ -104,6 +86,42 @@ fun ProvideInputDate(
...
+    return if (uiModel.selectableDates == null) {
+        if (uiModel.allowFutureDates == true) {
+            IntRange(1924, 2124)
+        } else {
+            (
+                IntRange(
+                    1924,
+                    Calendar.getInstance()[Calendar.YEAR],
+                )
+                )
```
Comment (andresmr) [inline form/src/main/java/org/dhis2/form/ui/provider/inputfield/DateProvider.kt:116]: is this properly formatted?

File: form/src/main/java/org/dhis2/form/data/EnrollmentRepository.kt
Comment (andresmr) [inline form/src/main/java/org/dhis2/form/data/EnrollmentRepository.kt:1]: What do you think about testing the new implementations of this class?

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialRepositoryImpl.java (hunk 1 above)
Comment (xavimolloy) [reply app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialRepositoryImpl.java:347]: the new null param is correct but I'm not sure why KtlintFormat didn't catch that, I'll update it

**Review by andresmr — APPROVED** [review]

**Review by mmmateos — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Pull request merged**